pytorch_lightning/trainer/training_io.py

```diff
@@ -306,42 +306,43 @@ def restore(self, checkpoint_path: str, on_gpu: bool):
         # load training state (affects trainer only)
         self.restore_training_state(checkpoint)
 
-    def dump_checkpoint(self):
+    def dump_checkpoint(self, weights_only: bool = False):
         checkpoint = {
             'epoch': self.current_epoch + 1,
             'global_step': self.global_step + 1,
         }
 
-        if self.checkpoint_callback is not None and self.checkpoint_callback is not False:
-            checkpoint['checkpoint_callback_best'] = self.checkpoint_callback.best
+        if not weights_only:
+            if self.checkpoint_callback is not None and self.checkpoint_callback is not False:
```

Comment:
```suggestion
            if self.checkpoint_callback:
```

Reply:
I didn't want to change parts of the code not directly related to the PR (to avoid creating a too complex PR), but this change is indeed a relief. :)

Reply:
GH marked it as changing part anyway...